shared/serial_port_settings.py

```diff
@@ -178,7 +179,21 @@ def edit_page(self, tab: str):
 
         # button
         self.save_button = CTkButton(self.edit_region, text="Save", command=self.save, height=14)
-        self.save_button.grid(row=11, column=2, padx=20, pady=40, sticky="ns")
+        self.save_button.grid(row=11, column=2, padx=20, pady=5, sticky="ns")
+
+        self.test_button = CTkButton(self.edit_region, text="Test Read", command=self.test_read, height=14)
+        self.test_button.grid(row=12, column=2, padx=20, pady=5, sticky="ns")
+
+        self.data_text = CTkLabel(self.edit_region, height=25, width=50, text="Waiting for Data...")
+        self.data_text.grid(row=13, column=2, columnspan=3, padx=20, pady=3, sticky="ew")
```

Comment:
This code is used to do a test reading of the serial device. Does it work more than once to allow for setting changes if needed? It looks like it may only attempt to read from the device exactly once.